CMakeLists.txt

```diff
@@ -771,14 +771,20 @@ pkg_check_modules(ZSTD QUIET libzstd>=1.4.0)
 check_include_file(zstd.h HAVE_ZSTD_H)
 if(ZSTD_FOUND AND HAVE_ZSTD_H)
   message(STATUS "Using shared zstd version: " ${ZSTD_VERSION})
+  set(ZSTD_LIB_NAME zstd)
 else()
   check_vendoring_approved(zstd)
   if(ZSTD_FOUND AND NOT HAVE_ZSTD_H)
     message(STATUS "Shared zstd found but lacks headers, falling back to the static library")
   else()
     message(STATUS "Shared zstd not found, falling back to the static library")
   endif()
-  add_subdirectory(Externals/zstd)
+  set(ZSTD_BUILD_SHARED OFF)
+  set(ZSTD_BUILD_PROGRAMS OFF)
+  add_subdirectory(Externals/zstd/zstd/build/cmake)
+  set(ZSTD_LIB_NAME libzstd_static)
+  dolphin_disable_warnings_msvc(libzstd_static)
+  include_directories(Externals/zstd/zstd/lib)
```

Comment:
I realize we're not really consistent with this, but I think it's better to have a CMakeLists.txt at `Externals/zstd` (ie, outside the actual submodule, where the VS project file also is) and call into that. Makes it clearer what flags we set for each dependency.

Reply:
Where should zstd_lib_name be set? Should it still be set like I’ve done it? A bit annoying that the name changes

Reply:
I moved `include_directories(zstd/lib)` into the zstd-local CMakeLists.txt and now that causes the zstd.h not to be exposed to dependents :/

Reply:
tbh i've exhausted my patience with cmake trying to get it to work properly

Reply:
Fair enough, I suppose.